libnmstate/state.py

```diff
@@ -399,6 +399,18 @@ def merge_interfaces(self, other_state):
             dict_update(other_state.interfaces[name], self.interfaces[name])
             self._ifaces_state[name] = other_state.interfaces[name]
 
+    def remove_unknown_interfaces(self):
+        """
+        Remove unknown type interfaces
+        """
+        unknown_ifaces = [
+            ifname
+            for ifname, ifstate in self.interfaces.items()
+            if ifstate.get(Interface.TYPE) == InterfaceType.UNKNOWN
+        ]
+        for unknown_iface in unknown_ifaces:
+            del self.interfaces[unknown_iface]
```

Comment:
Can we use ‘create_state’ factory instead of this func?
It feels wrong to delete items from a list currently iterated and it is a bit odd we remove the interfaces only now.
Another option is to fail the validation.

Or are these unknown ifaces a result of the previous processing?

Reply:
This unknown interface is the loopback interface.  Please refer to https://github.com/nmstate/nmstate/pull/620#issuecomment-560243752

The code is not deleting item from interating list. It stored a list of interface need to be deleted
in `unknown_ifaces`. Then delete it from `self.interfaces()`.